### PR TITLE
[nan-448] allow enterprise to sign up with a known token

### PR DESCRIPTION
**Pull request by khaliqgant**
## Describe your changes
1. Enterprise customer sets `NANGO_ADMIN_INVITE_TOKEN` to a known value
2. On creation of their Nango instance they navigate to `{customer}.nango.com/signup/{known-token}`
3. They enter their name and password, get directed to the signin page
4. Signin

Also disables the signup page for enterprise
Adds more permissive route access for local

## Issue ticket number and link
NAN-448

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

File: packages/shared/lib/services/user.service.ts [hunk 1]
```diff
@@ -136,6 +137,18 @@ class UserService {
     async getInvitedUserByToken(token: string): Promise<InviteUser | null> {
         const date = new Date();
 
+        if (isEnterprise() && process.env['NANGO_ADMIN_INVITE_TOKEN'] === token) {
+            return {
```
Comment (TBonnin) [inline packages/shared/lib/services/user.service.ts:141]: can we return null instead of a empty user?

File: packages/webapp/src/App.tsx [hunk 2]
```diff
@@ -140,17 +140,17 @@ const App = () => {
                         <Route path="/:env/project-settings" element={<ProjectSettings />} />
                     </Route>
                     <Route path="/auth-link" element={<AuthLink />} />
-                    {(isCloud() || isEnterprise()) && (
+                    {(isCloud() || isEnterprise() || isLocal()) && (
```
Comment (TBonnin) [inline packages/webapp/src/App.tsx:143]: when is this condition not true anymore now?

**Review by bodinsamuel — APPROVED** [review]

File: packages/shared/lib/services/user.service.ts (hunk 1 above)
Comment (khaliqgant) [reply packages/shared/lib/services/user.service.ts:141]: Well, if we return null then the path would think there is no actual user so it is an empty user to get around that.

File: packages/webapp/src/App.tsx (hunk 2 above)
Comment (khaliqgant) [reply packages/webapp/src/App.tsx:143]: hosted

File: packages/shared/lib/services/user.service.ts (hunk 1 above)
Comment (TBonnin) [reply packages/shared/lib/services/user.service.ts:141]: ok. can we add a comment to explain that in the code. I am sure if I read it again in 3 months I would have forgotten about it

File: packages/webapp/src/App.tsx (hunk 2 above)
Comment (TBonnin) [reply packages/webapp/src/App.tsx:143]: we are gonna need an enum at some point to track all different type of installations :D

**Review by TBonnin — APPROVED** [review]

File: packages/shared/lib/services/user.service.ts (hunk 1 above)
Comment (bodinsamuel) [reply packages/shared/lib/services/user.service.ts:141]: I think once again, a good env var (or a computed env var) would be easier to follow. Ultimately those are all feature flags that could be centralized.

**Comment by khaliqgant** [comment]
Thanks both. I'll make a Linear issue around flag consolidation to clean up this enterprise logic.

**Comment by khaliqgant** [comment]
> Thanks both. I'll make a Linear issue around flag consolidation to clean up this enterprise logic.

https://linear.app/nango/issue/NAN-533/clean-up-isenterprise-logic-in-favor-of-enumenv-variable

**Pull request merged**